docs/root/intro/arch_overview/advanced/attributes.rst

```diff
@@ -0,0 +1,153 @@
+.. _arch_overview_attributes:
+
+Attributes
+==========
+
+Attributes refer to contextual properties provided by Envoy during request and
+connection processing. They are named by a dot-separated path (e.g.
+`request.path`), have a fixed type (e.g. `string` or `int64`), and may be
+absent or present depending on the context. Attributes are exposed to CEL
+runtime in :ref:`RBAC filter <arch_overview_rbac>`, as well as Wasm extensions
+via `get_property` ABI method.
+
+Request attributes
+------------------
+
+The following request attributes are generally available upon initial request
+processing, which makes them suitable for RBAC policies:
+
+.. csv-table::
+   :header: Attribute, Type, Description
+   :widths: 1, 1, 4
+
+   request.path, string, The path portion of the URL
+   request.url_path, string, The path portion of the URL without the query string
+   request.host, string, The host portion of the URL
+   request.scheme, string, The scheme portion of the URL e.g. "http"
+   request.method, string, Request method e.g. "GET"
+   request.headers, "map<string, string>", All request headers indexed by the lower-cased header name
+   request.referer, string, Referer request header
+   request.useragent, string, User agent request header
+   request.time, timestamp, Time of the first byte received
+   request.id, string, Request ID
+   request.protocol, string, Request protocol e.g. "HTTP/2"
+
+Additional attributes are available once the request is finished:
+
+.. csv-table::
+   :header: Attribute, Type, Description
+   :widths: 1, 1, 4
+
+   request.duration, duration, Total duration of the request
+   request.size, int64, Size of the request body. Content length header is used if available.
```

Comment:
Why is this signed? Again, see above comments on being explicit around typing.

Reply:
Basic ints are preferable due to usability. Constants are typed as ints, not uints. All ints are 64-bit width in CEL.